citation/graphviz/data.py

```diff
@@ -62,12 +62,17 @@ def generate_link_candidates(filter_criteria):
 
 def get_network_default_filter(group_by):
     if group_by == NetworkGroupBYType.SPONSOR.value:
-        sponsors = Publication.api.primary(status="REVIEWED").values('sponsors__name').order_by('sponsors__name'). \
-                       annotate(count=Count('sponsors__name')).values('count', 'sponsors__name').order_by('-count')[:10]
+        # FIXME: this should be abstracted into a method like Publication.api.sponsors_with_count(number=10)
+        sponsors = Publication.api.primary(status="REVIEWED").values('sponsors__name').order_by(
+            'sponsors__name').annotate(count=Count('sponsors__name')).values(
+                'count', 'sponsors__name').order_by('-count')[:10]
         return [sponsor['sponsors__name'] for sponsor in sponsors]
     else:
-        tags = Publication.api.primary(status="REVIEWED").values('tags__name').order_by('tags__name'). \
-                   annotate(count=Count('tags__name')).values('count', 'tags__name').order_by('-count')[:10]
+        # FIXME: this should be abstracted into a method in Publication.api, e.g.,
+        # Publication.api.tags_with_count(number=10)
```

Comment:
I was thinking for creating one method like 
fetch_top_records(identifier= None, number=10)
Publication.api.fetch_top_records('tags',50)

Reply:
That sounds reasonable, though there are assumptions on the `__name` attribute? Maybe `attribute` instead of `identifier`, something like `get_top_records(attribute='tags__name', number=10)`